### ISSUE TITLE
NSID for dig

### ISSUE THREAD

**Issue by chellige** [kellyjonbrazil/jc#409]
Hello Kelly,
Your parsers are brilliant and very much appreciated. I'm a heavy user of your dig parser and would like to request one additional data option.  Most large DNS operations use virtual IP's for their advertised resolver and and behind that are countless physical resolvers doing the work. There is often a need to see which physical server is responding behind the VIP and the best way to do that is with the dig +nsid option. This adds one line to the response letting you know which server provided the answer

dig google.com +nsid
...
;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
; NSID: 41 6e 73 77 65 72 78 5f 4f 6e 5f 73 6e 6c 6f 63 61 2d 72 64 6e 73 2d 30 35 ("Answerx_On_snloca-rdns-05")
;; QUESTION SECTION:
...
It would be super useful if you could add support for this dig option.  I don't really see the need for the hex in the middle. Just need that answer at the end of the line between the parens.

Thanks again

**Comment by kellyjonbrazil** [comment]
Thanks for the suggestion! Doesn’t sound like it would be too hard to add that. I’ll look at that for the next release.

**Comment by chellige** [comment]
Sweet! Thanks again for all you do.

**Comment by kellyjonbrazil** [comment]
What version of `dig` are you using? Looks like the version I have must be pretty ancient and doesn't seem to support that option. (v9.11)

If you could provide me a couple of full `dig` outputs with the `+nsid` option that would help. Otherwise I can install a newer version and run some queries myself.

**Comment by chellige** [comment]
Hello Kelly. Thanks for working on this. I believe your version is fine. The issue is that not all systems are configured this way. Try the following query:
```
# dig -v
DiG 9.11.4-P2-RedHat-9.11.4-26.P2.el7_9.13

[root@ns1 ~]# dig +nsid @dns.google mail.google.com
; <<>> DiG 9.11.4-P2-RedHat-9.11.4-26.P2.el7_9.13 <<>> +nsid @dns.google mail.google.com
; (4 servers found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 46011
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
; NSID: 67 70 64 6e 73 2d 6f 72 64 ("gpdns-ord")
;; QUESTION SECTION:
;mail.google.com.               IN      A

;; ANSWER SECTION:
mail.google.com.        300     IN      A       142.250.191.197

;; Query time: 25 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Sat Jun 03 15:29:46 UTC 2023
;; MSG SIZE  rcvd: 73

**Comment by kellyjonbrazil** [comment]
I added `+nsid` support in the `dev` branch. You can test by downloading the parser file and putting it into your plugin directory:

https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/dig.py

- Linux/unix: `$HOME/.local/share/jc/jcparsers`
- macOS: `$HOME/Library/Application Support/jc/jcparsers`
- Windows: `$LOCALAPPDATA\jc\jc\jcparsers`

Let me know if you run into any issues!

**Comment by chellige** [comment]
Beautiful! Thanks so much for all you do Kelly. NSID is working great.
```
# dig google.com +nsid | jc --dig | jq .
[
  {
    "id": 58700,
    "opcode": "QUERY",
    "status": "NOERROR",
    "flags": [
      "qr",
      "rd",
      "ra"
    ],
    "query_num": 1,
    "answer_num": 1,
    "authority_num": 0,
    "additional_num": 1,
    "opt_pseudosection": {
      "edns": {
        "version": 0,
        "flags": [],
        "udp": 512
      },
      "nsid": "Answerx_On_snloca-rdns-03"
    },
    "question": {
      "name": "google.com.",
      "class": "IN",
      "type": "A"
    },
    "answer": [
      {
        "name": "google.com.",
        "class": "IN",
        "type": "A",
        "ttl": 64,
        "data": "142.250.72.238"
      }
    ],
    "query_time": 31,
    "server": "71.10.216.1#53(71.10.216.1)",
    "when": "Mon Jun 05 23:54:39 UTC 2023",
    "rcvd": 84,
    "when_epoch": 1686009279,
    "when_epoch_utc": 1686009279
  }
]

**Comment by kellyjonbrazil** [comment]
You're welcome! I'll leave this issue open until I release v1.23.3.

**Comment by kellyjonbrazil** [comment]
Released in v1.23.3
